tokio/src/io/interest.rs

```diff
@@ -44,6 +44,10 @@ impl Interest {
     /// Writable interest includes write-closed events.
     pub const WRITABLE: Interest = Interest(mio::Interest::WRITABLE);
 
+    /// Returns a `Interest` set representing priority completion interests.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    pub const PRIORITY: Interest = Interest(mio::Interest::PRIORITY);
```

Comment:
Most types from `tokio::net` have `ready()` and `try_io()` methods which take `Interest` as an argument, we should make sure they are well-behaved with this new type of interest. I think you will have to make some extra modifications to `src/io/ready.rs` and/or `src/io/interest.rs` to make those methods work.

Reply:
I think I've got them all. The one place I'm not entirely sure about is in `tokio/src/runtime/io/mod.rs`

```rust
impl Direction {
    pub(super) fn mask(self) -> Ready {
        match self {
            Direction::Read => Ready::READABLE | Ready::READ_CLOSED,
            Direction::Write => Ready::WRITABLE | Ready::WRITE_CLOSED,
        }
    }
}
```

I left this untouched, because it really seems to be about reading and writing here.

Reply:
Yes, that's for AsyncRead/AsyncWrite only, thanks.